buffer/src/main/java/io/netty5/buffer/api/bytebuffer/NioBuffer.java

```diff
@@ -1137,23 +1148,30 @@ private RuntimeException checkWriteState(IndexOutOfBoundsException ioobe, int of
 
     private RuntimeException readAccessCheckException(int index, int size) {
         if (rmem == CLOSED_BUFFER) {
-            throw bufferIsClosed(this);
+            return bufferIsClosed(this);
         }
         return outOfBounds(index, size);
     }
 
-    private RuntimeException writeAccessCheckException(int index, int size) {
+    private void writeAccessCheckException(int index, int size, boolean mayExpand) {
```

Comment:
nit: consider renaming to something like `checkWriteAccess(...)` as it is throwing now directly